packages/openapi3/test/emit-openapi.test.ts

```diff
@@ -1,79 +1,136 @@
+            Yes: "Yes",
+            YES: "YES",
+            yEs: "yEs",
+            n: "n",
+            N: "N",
+            no: "no",
+            No: "No",
+            NO: "NO",
+            nO: "nO",
+            "true": "true",
+            True: "True",
+            TRUE: "TRUE",
+            tRUE: "tRUE",
+            "false": "false",
+            False: "False",
+            FALSE: "FALSE",
+            fALSE: "fALSE",
+            on: "on",
+            On: "On",
+            ON: "ON",
+            oN: "oN",
+            off: "off",
+            Off: "Off",
+            OFF: "OFF",
+            oFF: "oFF"
+          }
+          `);
+      expect(content).toBe(`openapi: 3.0.0
+    info:
+      title: (title)
+      version: 0.0.0
+    tags: []
+    paths: {}
+    components:
+      schemas:
+        TestEnum:
+          type: string
+          enum:
+            - 'y'
+            - 'Y'
+            - 'yes'
+            - 'Yes'
+            - 'YES'
+            - yEs
+            - 'n'
+            - 'N'
+            - 'no'
+            - 'No'
+            - 'NO'
+            - nO
+            - 'true'
+            - 'True'
+            - 'TRUE'
+            - tRUE
+            - 'false'
+            - 'False'
+            - 'FALSE'
+            - fALSE
+            - 'on'
+            - 'On'
+            - 'ON'
+            - oN
+            - 'off'
+            - 'Off'
+            - 'OFF'
+            - oFF
+    `);
+    });
+
+    interface ServiceNameCase {
+      description: string;
+      code: string;
+      outputFilePattern: string;
+      expectedOutputFiles: string[];
+    }
+    it.each([
+      // {service-name} cases
+      {
+        description: "{service-name} for one service",
+        code: "@service namespace AAA { model M {a: string} }",
+        outputFilePattern: "{service-name}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("AAA.yaml")],
+      },
+      {
+        description: "{service-name} for multiple services",
+        code:
+          "@service namespace AAA { model M {a: string} }" +
+          "@service namespace BBB { model N {b: string} }",
+        outputFilePattern: "{service-name}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("AAA.yaml"), resolveVirtualPath("BBB.yaml")],
+      },
+      // {service-name-if-multiple} cases
+      {
+        description: "{service-name-if-multiple} for one service",
+        code: "@service namespace AAA { model M {a: string} }",
+        outputFilePattern: "{service-name-if-multiple}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("yaml")],
+      },
+      {
+        description: "{service-name-if-multiple} for multiple services",
+        code:
+          "@service namespace AAA { model M {a: string} }" +
+          "@service namespace BBB { model N {b: string} }",
+        outputFilePattern: "{service-name-if-multiple}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("AAA.yaml"), resolveVirtualPath("BBB.yaml")],
```

Comment:
the test should probabluy just be the one calling `resolveVirtualPath`

Reply:
But we have multiple outputs, not sure what are you saying

Reply:
just remove the function calls in there and call the funcgtion in the test logic instead